TAScheduler/test/test_assignCourseInstructor.py

```diff
@@ -0,0 +1,148 @@
+from django.test import TestCase
+from ..models import Account
+from ..models import Course
+from ..domain.CommandHandler import CommandHandler
+from ..domain.ErrorMessages import ErrorMessages
+
+class EditTestCase(TestCase):
+    def setUp(self):
+        Account.objects.create(act_email='super@email.com', act_fname='Super', act_lname='Visor',
+                               act_password='4444', act_address='123 main st.', act_phone='444-444-4444', role_id=4,
+                               act_officehours="11-12", act_officelocation="EMS 124")
+        Account.objects.create(act_email='admin@email.com', act_fname='Admin', act_lname='Istrator',
+                               act_password='3333', act_address='10 First st.', act_phone='333-333-3333', role_id=3,
+                               act_officehours="9-10", act_officelocation="EMS 123")
+        self.inst1 = Account.objects.create(act_email='instructor@email.com', act_fname='Instr', act_lname='Uctor',
+                               act_password='2222', act_address='17 North st.', act_phone='222-222-2222', role_id=2,
+                               act_officehours="10-11", act_officelocation="EMS 122")
+        self.inst2 = Account.objects.create(act_email='instructor2@email.com', act_fname='Instr2', act_lname='Uctor2',
+                               act_password='22222', act_address='172 North st.', act_phone='222-222-2223', role_id=2,
+                               act_officehours="10-12", act_officelocation="EMS 120")
+        Account.objects.create(act_email='ta@email.com', act_fname='TA', act_lname='Student',
+                               act_password='1111', act_address='24 Second St', act_phone='111-111-1111', role_id=1,
+                               act_officehours="1-2", act_officelocation="EMS 121")
+
+        Course.objects.create(course_id='535', course_name='Algorithms', instructor=None)
+        Course.objects.create(course_id='537', course_name='OS', instructor=self.inst2)
+
+
+    def test_without_login__assignment_fails(self):
+        ch = CommandHandler()
+        msg = ch.ProcessCommand("assign course instructor Algorithms instructor@email.com")
+        self.assertEqual(msg, 'Must be logged in as an Administrator or Supervisor')
+
+    def test_supervisor_assigns_valid_instructor_and_valid_class__then_assign_succeeds(self):
+        ch = CommandHandler()
+        ch.ProcessCommand("login super@email.com 4444")
```

Comment:
In the setup we should do the login. We can have different tests classes for specific roles. For example, we could do SupervisorEditTestCase, AdminEditTestCase, etc. Just helps the redundancy

Reply:
Good idea.  we can probably wait until next sprint to clean  that up.  Future tests can follow the better pattern though.

Reply:
Unless Rachel is really ambitious!